pandas/tests/extension/test_categorical.py

```diff
@@ -93,7 +93,8 @@ class TestConstructors(base.BaseConstructorsTests):
 
 
 class TestReshaping(base.BaseReshapingTests):
-    pass
+    def test_concat_with_reindex(self, data):
+        pytest.xfail(reason="Deliberately upcast to object?")
```

Comment:
Do we have an issue for this? (it seems we shouldn't cast to object when only needing to append NaNs?)